Common/include/adt/CADTElemClass.hpp

```diff
@@ -28,238 +26,9 @@
-private:
-  vector<su2double>     coorPoints;    /*!< \brief Vector, which contains the coordinates
-                                                   of the points in the ADT. */
-  vector<unsigned long> localPointIDs; /*!< \brief Vector, which contains the local point ID's
-                                                   of the points in the ADT. */
-  vector<int>           ranksOfPoints; /*!< \brief Vector, which contains the ranks
-                                                   of the points in the ADT. */
-public:
-  /*!
-   * \brief Constructor of the class.
-   * \param[in] nDim       Number of spatial dimensions of the problem.
-   * \param[in] nPoints    Number of local points to be stored in the ADT.
-   * \param[in] coor       Coordinates of the local points.
-   * \param[in] pointID    Local point IDs of the local points.
-   * \param[in] globalTree Whether or not a global tree must be built. If false
-                           a local ADT is built.
-   */
-  CADTPointsOnlyClass(unsigned short nDim,
-                      unsigned long  nPoints,
-                      su2double      *coor,
-                      unsigned long  *pointID,
-                      const bool     globalTree);
-
-  /*!
-   * \brief Function, which determines the nearest node in the ADT for the given coordinate.
-   * \note This simply forwards the call to the implementation function selecting the right
-   *       working variables for the current thread.
-   * \param[in]  coor    Coordinate for which the nearest node in the ADT must be determined.
-   * \param[out] dist    Distance to the nearest node in the ADT.
-   * \param[out] pointID Local point ID of the nearest node in the ADT.
-   * \param[out] rankID  Rank on which the nearest node in the ADT is stored.
-   */
-  inline void DetermineNearestNode(const su2double *coor,
-                                   su2double       &dist,
-                                   unsigned long   &pointID,
-                                   int             &rankID) {
-    const auto iThread = omp_get_thread_num();
-    DetermineNearestNode_impl(FrontLeaves[iThread], FrontLeavesNew[iThread],
-                              coor, dist, pointID, rankID);
-  }
-
-  /*!
-   * \brief Default constructor of the class, disabled.
-   */
-  CADTPointsOnlyClass() = delete;
-
-private:
-  /*!
-   * \brief Implementation of DetermineNearestNode.
-   * \note Working variables (first two) passed explicitly for thread safety.
-   */
-  void DetermineNearestNode_impl(vector<unsigned long>& frontLeaves,
-                                 vector<unsigned long>& frontLeavesNew,
-                                 const su2double *coor,
-                                 su2double       &dist,
-                                 unsigned long   &pointID,
-                                 int             &rankID) const;
-};
+#include "./CADTBaseClass.hpp"
+#include "./CBBoxTargetClass.hpp"
+#include "../omp_structure.hpp"
```

Comment:
There was quite some room for cutting down includes. I tried to include the bare minimum in the new files